homeassistant/components/tplink/sensor.py

```diff
@@ -28,6 +29,9 @@ class TPLinkSensorEntityDescription(
 ):
     """Base class for a TPLink feature based sensor entity description."""
 
+    #: Optional callable to convert the value
+    convert_fn: Callable[[Any], Any] | None = None
```

Comment:
See comment about using native units and letting ha do the conversion via `suggested_unit_of_measurement`

https://github.com/home-assistant/core/pull/136510/files